core/primitives/src/utils.rs

```diff
@@ -98,17 +112,43 @@ impl<T> MaybeValidated<T> {
     /// Applies function to the payload (whether it’s been validated or not) and
     /// returns new object with result of the function as payload.  Validated
     /// state is not changed.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use near_primitives::utils::MaybeValidated;
+    ///
+    /// let value = MaybeValidated::from(42);
+    /// assert_eq!("42", value.map(|v| v.to_string()).into_inner());
+    /// ```
     pub fn map<U, F: FnOnce(T) -> U>(self, validator: F) -> MaybeValidated<U> {
         MaybeValidated { validated: self.validated, payload: validator(self.payload) }
     }
 
     /// Returns a new object storing reference to this object’s payload.  Note
     /// that the two objects do not share the validated state so calling
     /// `validate_with` on one of them does not affect the other.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use near_primitives::utils::MaybeValidated;
+    ///
+    /// let value = MaybeValidated::from(42);
+    /// let value_as_ref = value.as_ref();
+    /// assert_eq!(Ok(true), value_as_ref.validate_with::<(), _>(|&&v| Ok(v == 42)));
+    /// assert!(value_as_ref.is_validated());
+    /// assert!(!value.is_validated());
+    /// ```
```

Comment:
I tend to think that doctests are not worth it for non-library crates in general. The problem with doctests is that they are notoriously slow to compile and run, as each doctest is compiled as a separate binary. For example, in rust-analyzer, removing `doctest = false` from Cargo.tomls increases the wall-clock time for the testsuite from 5 seconds to 10 seconds, and ra doesn't even have doctets.

OTOH, our tests are already pretty slow, and we dont' add `doctest=false` to cargo.tomls, so this isn't a regression for nearcore. So, no strong opinion here!

Reply:
I don’t have an opinion either way. Wasn’t really aware that there was noticeable compile-time cost. I’m happy to rewrite it as `[test]`.

Reply:
Nah, I guess let's leave it as is -- we do seem to have some doctests scattered elsewhere. Although, once #4490 is fixed, it makes sense to quantify how bad are doctests for our case. Maybe by that time https://github.com/rust-lang/rust/issues/75341 is fixed, which would be ideal :)